osv/models.py

```diff
@@ -352,9 +352,19 @@ def _pre_put_hook(self):
     self.ecosystem = list(ecosystems_set)
     self.ecosystem.sort()
 
-    self.purl = list({
-        pkg.package.purl for pkg in self.affected_packages if pkg.package.purl
-    })
+    def with_and_without_qualifiers(affected_packages) -> list[str]:
```

Comment:
Can you pull this up into a module level function? 

And name it `_get_purl_indexes` to make this more clear what it's for.